pkg/executables/kubectl.go

```diff
@@ -345,21 +345,22 @@ func (k *Kubectl) ValidateControlPlaneNodes(ctx context.Context, cluster *types.
 
 func (k *Kubectl) ValidateWorkerNodes(ctx context.Context, cluster *types.Cluster, clusterName string) error {
 	logger.V(6).Info("waiting for nodes", "cluster", clusterName)
-	md, err := k.GetMachineDeployment(ctx, cluster, clusterName, WithCluster(cluster), WithNamespace(constants.EksaSystemNamespace))
+	deployments, err := k.GetMachineDeployments(ctx, WithCluster(cluster), WithNamespace(constants.EksaSystemNamespace))
 	if err != nil {
 		return err
 	}
+	for _, machineDeployment := range deployments {
```

Comment:
Should we be looping through the worker node groups and iterating through that list and check for machine groups accordingly? That can be done in a followup because this is not controller code so this use case would be a lot more rare.